frontend/src/components/RequestTargetInputs.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+  <div class="request-target-inputs flex flex-col gap-7">
+    <Selector
+      v-model="selectedTargetChain"
+      label="To"
+      name="targetChain"
+      :options="targetChains"
+      placeholder="Target Rollup"
+      required
+      @input="updateTargetChain"
+    />
+    <div class="flex flex-row gap-5">
+      <div class="flex-[3_3_0%] flex flex-col items-end">
+        <TextInput :model-value="props.modelValue.amount" placeholder="0.00" disabled />
+      </div>
+      <div class="flex-[2_2_0%]">
+        <Selector
+          id="token"
+          :model-value="props.modelValue.token"
+          :options="[]"
+          placeholder="Token"
+          disabled
+        />
+      </div>
+    </div>
+    <TextInput
+      v-model="selectedTargetAddress"
+      name="toAddress"
+      placeholder="address"
+      required
+      @input="updateTargetAddress"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { storeToRefs } from 'pinia';
+import { computed, ref, watch } from 'vue';
+
+import Selector from '@/components/inputs/Selector.vue';
+import TextInput from '@/components/inputs/TextInput.vue';
+import { useChainSelection } from '@/composables/useChainSelection';
+import { useConfiguration } from '@/stores/configuration';
+import { useEthereumProvider } from '@/stores/ethereum-provider';
+import type { Chain } from '@/types/data';
+import type { RequestFormResult, SelectorOption } from '@/types/form';
+
+interface Props {
+  modelValue: RequestFormResult;
+}
+
+interface Emits {
+  (e: 'update:modelValue', value: RequestFormResult): void;
+}
+
+const props = defineProps<Props>();
+const emits = defineEmits<Emits>();
+
+const configuration = useConfiguration();
+const ethereumProvider = useEthereumProvider();
+
+const { provider } = storeToRefs(ethereumProvider);
+const { chains } = storeToRefs(configuration);
+
+const selectedTargetChain = ref<SelectorOption<Chain> | null>(null);
+const selectedTargetAddress = ref('');
+
+const selectedSourceChain = computed(() => props.modelValue.sourceChain);
```

Comment:
I guess this needs to change to a standalone property if you agree to rework the `v-model`.